onnxscript/rewriter/ort_fusions/cos_sin_cache.py

```diff
@@ -170,11 +170,8 @@ def rewrite(
 
 
 def fuse_cos_sin_cache(model: ir.Model, debug: bool = False) -> int:
-    count = cos_sin_cache_rules.apply_to_model(model)
-    if count == 0 and debug:
-        tracer = pattern.MatchingTracer()
-        cos_sin_cache_rules.apply_to_model(model, tracer=tracer)
-        tracer.report()
+    fuse_cos_sin_cache = _fusion_utils.apply_fusion_rules(cos_sin_cache_rules)
+    count = fuse_cos_sin_cache(model, debug=debug)
```

Comment:
The remove_unused_nodes part below is no longer necessary, because it has been made automatic (based on the rule's settings ... so, it will be called as the last step in apply_fusion_rules.) So, that should simply this part. Feel free to do this as part of this or as a separate PR, as convenient.